dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/OSUtils.java

```diff
@@ -156,65 +148,75 @@ public static double cpuUsage() {
         return Double.parseDouble(df.format(cpuUsage));
     }
 
-    public static List<String> getUserList() {
+    /**
+     * Determine if the tenantCode exists in the system
+     * @return boolean
+     **/
+    public static boolean isExistTenantCode(String tenantCode){
+
         try {
             if (SystemUtils.IS_OS_MAC) {
-                return getUserListFromMac();
+                return existTenantCodeInMac(tenantCode);
             } else if (SystemUtils.IS_OS_WINDOWS) {
-                return getUserListFromWindows();
+                return existTenantCodeInWindows(tenantCode);
             } else {
-                return getUserListFromLinux();
+                return existTenantCodeInLinux(tenantCode);
             }
         } catch (Exception e) {
             logger.error(e.getMessage(), e);
         }
 
-        return Collections.emptyList();
+        return false;
+
     }
 
     /**
-     * get user list from linux
+     * whether the user exists in linux
      *
-     * @return user list
+     * @return boolean
      */
-    private static List<String> getUserListFromLinux() throws IOException {
-        List<String> userList = new ArrayList<>();
-
-        try (BufferedReader bufferedReader = new BufferedReader(
-                new InputStreamReader(new FileInputStream("/etc/passwd")))) {
-            String line;
-
-            while ((line = bufferedReader.readLine()) != null) {
-                if (line.contains(":")) {
-                    String[] userInfo = line.split(":");
-                    userList.add(userInfo[0]);
-                }
+    private static boolean existTenantCodeInLinux(String tenantCode) throws IOException {
+        try{
+            String result = exeCmd("id "+ tenantCode);
```

Comment:
I think we can provide a switch param in ```common.properties``` to determine whether to use distributed authentication. If it is closed, leave it as it is. If it's ```on```, then use this function. And also we should put this change in the document so that users can chose it according to their scene. What do you think?

Reply:
> This command `id ${tennantCode}` may not be compatiable with users environment. It may throw `permission denied` exception when the user `dolphinscheduler` did not have execute permissions.

At present, it is verified that the id ${tennantCode} has permission to execute under any user, and does not need sudo

Reply:
> I think we can provide a switch param in `common.properties` to determine whether to use distributed authentication. If it is closed, leave it as it is. If it's `on`, then use this function. And also we should put this change in the document so that users can chose it according to their scene. What do you think?

This is a good idea because of the compatibility of the id command I can't test if all liunx versions are supported。

Reply:
> I think we can provide a switch param in `common.properties` to determine whether to use distributed authentication. If it is closed, leave it as it is. If it's `on`, then use this function. And also we should put this change in the document so that users can chose it according to their scene. What do you think?

hi, have time to see if the current logic is OK?

Reply:
> > I think we can provide a switch param in `common.properties` to determine whether to use distributed authentication. If it is closed, leave it as it is. If it's `on`, then use this function. And also we should put this change in the document so that users can chose it according to their scene. What do you think?
> 
> hi, have time to see if the current logic is OK?

I will take a look later.